### PR TITLE
ICU-22534 BRS#27 scrub closed issues (frontload)

### PR DESCRIPTION
**Pull request by sven-oly**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22534
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [x] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/util/ICUResourceBundleTest.java [hunk 1]
```diff
@@ -717,10 +717,6 @@ public void TestLocaleDisplayNames() {
         }
 
         Set<String> localLangExceptions = new HashSet<>();
-        if (logKnownIssue("cldrbug:17001", "No localized language name for nmg, vmw")) {
-            localLangExceptions.add("nmg");
-            localLangExceptions.add("vmw");
-        }
```
Comment (markusicu) [inline icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/util/ICUResourceBundleTest.java:723]: CLDR-17001 is done/fixed as of CLDR 44, but this test fails for language nmg.
The ticket only mentions language vmw, but this logKnownIssue was for this ticket and both languages.

https://github.com/unicode-org/icu/actions/runs/8147838298/job/22269422989?pr=2864
```
Error:  com.ibm.icu.dev.test.util.ICUResourceBundleTest.TestLocaleDisplayNames -- Time elapsed: 0.001 s <<< FAILURE!
java.lang.AssertionError: Could not get native localized language for nmg
	at org.junit.Assert.fail(Assert.java:89)
	at com.ibm.icu.dev.test.AbstractTestLog.errln(AbstractTestLog.java:50)
	at com.ibm.icu.dev.test.util.ICUResourceBundleTest.TestLocaleDisplayNames(ICUResourceBundleTest.java:738)
```

@DraganBesevic @pedberg-icu @srl295 @macchiati please advise.

**Review by markusicu — CHANGES_REQUESTED** [review]

File: icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/util/ICUResourceBundleTest.java (hunk 1 above)
Comment (sven-oly) [reply icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/util/ICUResourceBundleTest.java]: Thanks. I've just reverted this change.

Should we post another ticket for language nmg and change the reference?

Should I remove localLanExpections.add("vmw")?

Comment (markusicu) [reply icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/util/ICUResourceBundleTest.java]: > Should I remove localLanExpections.add("vmw")?

Yes, please try that.

> Should we post another ticket for language nmg and change the reference?

Probably. Let's see if we can get a little more feedback.

@macchiati CLDR-17001 was about a language which was at basic but didn't even have an English display name. It looks like @pedberg-icu and/or @DraganBesevic used the same ticket for a second language that didn't get mentioned in the ticket and apparently didn't get fixed.

**Comment by markusicu** [comment]
> Note: ...

@macchiati if you reply to PR comments by email, your comment is posted out of context which makes it hard to figure out who said what.

File: icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/util/ICUResourceBundleTest.java (hunk 1 above)
Comment (markusicu) [reply icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/util/ICUResourceBundleTest.java]: Looking at the CLDR 45 coverage levels: https://www.unicode.org/cldr/charts/45/supplemental/locale_coverage.html
- vmw Target Level = none, Computed Level = basic
- nmg Target Level = none, Computed Level = none

So maybe CLDR doesn't really care yet, and ICU should be able to live with these exceptions?

But if nmg is not yet at basic level, then why did it get added to ICU?

Comment (srl295) [reply icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/util/ICUResourceBundleTest.java]: > Looking at the CLDR 45 coverage levels: https://www.unicode.org/cldr/charts/45/supplemental/locale_coverage.html
> 
> * vmw Target Level = none, Computed Level = basic
> * nmg Target Level = none, Computed Level = none
> 
> So maybe CLDR doesn't really care yet, and ICU should be able to live with these exceptions?
> 
> But if nmg is not yet at basic level, then why did it get added to ICU?

There was an issue where ICU was only looking at new locales and not looking at coverageLevels.txt (that's the data file that should be used).

Comment (markusicu) [reply icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/util/ICUResourceBundleTest.java]: I submitted ICU-22681 “ICUResourceBundleTest.TestLocaleDisplayNames fails with ICU locales that are below CLDR basic coverage”

**Review by markusicu — APPROVED** [review]

**Comment by sven-oly** [comment]
Thanks for adding the ICU ticket, Markus.

**Comment by roubert** [comment]
This seems to have broken the tests, _CI-Exhaustive-Main (C: Linux Clang Exhaustive Tests (Ubuntu 22.04))_ now fails with this error message:

```
TestDataDrivenLikelySubtags {
addLikelySubtags('qaa-Cyrl-CH') should return '' but got 'qaa-Cyrl-CH'
minimizeSubtags('qaa-Cyrl-CH') should return '' but got 'qaa-Cyrl-CH'
minimizeSubtags('qaa-Cyrl-CH') favor script should return '' but got 'qaa-Cyrl-CH'

} ERRORS (3) in TestDataDrivenLikelySubtags (33ms) 
```

**Comment by markusicu** [comment]
> This seems to have broken the tests, _CI-Exhaustive-Main (C: Linux Clang Exhaustive Tests (Ubuntu 22.04))_ now fails with this error message:

@roubert and @sven-oly please work together to make the tests pass again.

**Comment by FrankYFTang** [comment]
The changes to [icu4c/source/test/intltest/loctest.cpp](https://github.com/unicode-org/icu/pull/2864/files#diff-725e2e6656f2cbe9f38588f8276eb69149f99d5c40ddcd4a3b81724d45cd3e8d) should be reverted I think

**Comment by roubert** [comment]
> @roubert and @sven-oly please work together to make the tests pass again.

Done, _CI-Exhaustive_ now passes again as of commit 89cf56333f3361470e4250e594c16c96b9634549.

**Pull request merged**